python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -225,57 +225,26 @@ def get_actions(self, data):
 
         manager = sgtk.bootstrap.ToolkitManager()
         manager.allow_config_overrides = False
+        manager.plugin_id = "basic.shotgun"
         manager.base_configuration = constants.BASE_CONFIG_URI
-        pcs = self._get_pipeline_configurations(
-            manager,
-            project_entity,
-        ) or [dict(id=None, name="Primary")]
-
-        self.logger.debug("Pipeline configurations found: %s" % pcs)
 
-        # We'll need to pass up the config names in order along with a dict of
-        # pc_name => commands.
-        pc_names = [pc["name"] for pc in pcs]
         all_actions = dict()
-        config_data = dict()
-
-        for pc in pcs:
-            self.logger.debug("Processing config: %s" % pc)
-
-            # The hash that acts as the key we'll use to look up our cached
-            # data will be based on the entity type and the pipeline config's
-            # descriptor uri. We can get the descriptor from the toolkit
-            # manager and pass that through along with the entity type from SG
-            # to the core hook that computes the hash.
-            manager.pipeline_configuration = pc["id"]
-            pc_descriptor = manager.resolve_descriptor(project_entity)
-
-            self.logger.debug("Resolved config descriptor: %r" % pc_descriptor)
-            pc_key = pc_descriptor.get_uri()
-
-            pc_data = dict()
-            pc_data["contents_hash"] = self._get_contents_hash(
-                pc_descriptor,
-                self._get_site_state_data(),
-            )
-            pc_data["lookup_hash"] = self._get_lookup_hash(
-                pc_key,
-                project_entity,
-                data["entity_type"],
-            )
-            pc_data["descriptor"] = pc_descriptor
-            pc_data["entity"] = pc
-            config_data[pc["id"]] = pc_data
+        all_pc_data = self._get_pipeline_configuration_data(
+            manager,
+            project_entity,
+            data,
+        )
 
         with self._db_connect() as (connection, cursor):
-            for pc in pcs:
+            for pc_id, pc_data in all_pc_data.iteritems():
+                pc = pc_data["entity"]
+
                 # The hash that acts as the key we'll use to look up our cached
                 # data will be based on the entity type and the pipeline config's
                 # descriptor uri. We can get the descriptor from the toolkit
                 # manager and pass that through along with the entity type from SG
                 # to the core hook that computes the hash.
-                pc_data = config_data[pc["id"]]
-                manager.pipeline_configuration = pc["id"]
+                manager.pipeline_configuration = pc_id
```

Comment:
Maybe I don't understand, but how can that be possible? If that's the case, how does the manager know what config to bootstrap? It can't just operate off of the base_configuration all the time, unless I'm misunderstanding the fundamentals (or they've changed).